worker/uniter/resolver.go

```diff
@@ -61,7 +61,7 @@ func (s *uniterResolver) NextOp(
 	remoteState remotestate.Snapshot,
 	opFactory operation.Factory,
 ) (operation.Operation, error) {
-	if remoteState.Life == life.Dead || localState.Stopped {
+	if remoteState.Life == life.Dead || localState.Removed {
```

Comment:
Is this correct, I don't know enough about the consequences to this...?

Reply:
Before this change, `stop` was the last hook that got executed before the uniter was signaled to die. So basically, once the unit is dying you would first hit [this](https://github.com/juju/juju/pull/11237/files/907b02253fbddceb29953517186e2da532ff73cd#diff-2e393918016123f29cf986dc718755d8R279) line which triggers the `stop` hook. Then you would enter `NextOp` because of [this](https://github.com/juju/juju/blob/907b02253fbddceb29953517186e2da532ff73cd/worker/uniter/operation/runhook.go#L257-L260) bit of logic (TLDR: asks uniter to figure out the next op) and exit with `ErrTerminate` to kill the uniter.

Now, the last hook in the chain is `remove` hence the need for the above change; otherwise the state machine would stop at `stop`.